### ISSUE TITLE
Release notes for 0.214

### ISSUE THREAD

**Issue by findepi** [prestodb/presto#11804]

**Comment by nezihyigitbasi** [comment]
```
Hive Changes
------------

* Add ``hive.s3.upload-acl-type`` config option to specify the canned ACL to use while uploading 
  files to S3.
```

**Comment by haozhun** [comment]
```
General Changes
---------------

* Fix memory leak related to query tracking in coordinator. Completed queries
  are supposed to be purged coordinator shortly after its completion. However,
  a regression in 0.213 broke this.

**Comment by anusudarsan** [comment]
```
General Changes
---------------
* Fix bug where the CLI throws a vague error ``Query gone(server restarted?)`` instead of actual ``Exception`` message.
```

**Comment by dain** [comment]
```
General Changes
---------------
* Remove experimental preallocated memory system, and the related configuration property ``experimental.preallocate-memory-threshold``.
* Add system to delay query execution until a minimum number of workers nodes are available.  If the requirement is not met within a time limit, the query is failed.  The minimum number of workers can be set with the ``query-manager.required-workers`` configuration property, and the max wait time can ``query-manager.required-workers-max-wait`` configuration property.
```

**Comment by ggreg** [comment]
## Andrii Rosa          
- [ ] all checked
-  11-08 08:23:17    4f679c640f Enable stats calculator by default
-  11-08 08:23:17    b5fb41f368 Ignore failures in stats and cost calculator
## Dain Sundstrom       
- [ ] all checked
-  11-02 10:27:14    744be58824 Count max code point once per batch for slice readers
-  11-04 13:07:47    0677563bad Simplify QueryStateMachine getQueryInfo
-  11-04 13:07:47    3c148f0d5e Use query id from session where available
-  11-04 13:07:47    3e2b9c11b3 Simplify and cleanup QueryMonitor code
-  11-04 13:07:47    71e4bd04e3 Replace QueryManager.getResourceGroup with BasicQueryInfo
-  11-04 13:07:47    8e33831a5e Remove unused QueryStateMachine failed factory methods
-  11-04 13:07:47    d367ac57bc Require resource group when creating QueryStateMachine
-  11-04 13:07:48    105a057fa6 Remove experimental preallocated memory system
-  11-04 13:07:48    60d59d8943 Force node manager refresh after TestingPrestoServer add catalog
-  11-04 13:07:48    88e8598ee1 Fix IntelliJ warnings
-  11-04 13:07:48    b77fa0bc74 Use PreparedQuery for QueryExecutionFactory createQueryExecution
-  11-04 13:07:48    f8578fbb5b Remove unused DataDefinitionExecutionFactory explain method
-  11-04 13:36:08    90dfc04acd Simplify and cleanup DiscoveryNodeManager
-  11-04 13:48:05    9408e074f3 Add abstraction to track minimum worker requirement
-  11-04 13:48:06    c00367d23c Add required minimum number of workers to start query
-  11-06 14:05:58    4d4b0bbe0a Fix leak of failed/aborted queries in coordinator
-  11-06 14:05:58    d1a93459a3 Fix warnings in StateMachine and TestStateMachine
-  11-06 14:05:58    d7780854c0 Remove unnecessary TaskInfo.isComplete() method
-  11-06 14:05:58    e3c5097d14 Ensure StateMachine listeners are called back on a different thread
-  11-06 16:19:48    afd294f3b9 Fix race condition in remote task final info visibility
-  11-06 16:19:48    fe743c0b18 Revert "Fix memory leak for failed queries"
-  11-06 16:21:09    b09ea18519 Improve GuardedBy coverage in SqlStageExecution
-  11-06 16:21:10    1fb288ab74 Increase timeout of TestDriver.testConcurrentClose
-  11-06 16:21:10    522d6fcee8 Null out executors in TestSqlStageExecution.tearDown
-  11-06 16:21:10    9011cfb776 Increase timeouts in TestStateMachine
-  11-06 16:21:10    d01145484b Only create FinalizerService executor if started
-  11-07 14:54:51    3c15bb126f Add unprocessed catalog properties to query events
## David Phillips       
- [ ] all checked
-  10-28 12:33:51    503f5ed3b0 Handle SQL execution timeouts for verifier rewrite
-  10-29 13:41:27    2a3a3e0d65 Clear update count on getMoreResults method call
-  10-29 17:36:15    1050b256a4 Categorize error for broken array_sort comparator
-  10-29 17:36:15    57b4aded20 Cleanup duplicate code in ArraySortComparatorFunction
-  10-30 10:01:56    2632c29f57 Update to Airbase 86
-  10-30 10:01:56    43ccfd542d Add whitespace around operators
-  10-30 10:01:56    a7b0b36488 Remove static imports for Optional
-  10-30 16:29:33    07aa4558ae Retry on Hive metastore connection errors
-  10-30 17:01:44    362a1f6d9d Cleanup code formatting
-  11-06 13:33:28    5180540829 Remove unused constructor for PrestoException
-  11-06 13:33:28    dddd9ea84e Simplify exception super constructor calls
-  11-06 14:41:19    d16fdc2875 Remove path from ConnectorSession
-  11-07 11:08:11    0b93d49ad6 Fix deployment of presto-jdbc artifact
-  11-08 13:20:39    df56cfaa4d Update AWS SDK to version 1.11.445
## Grzegorz Kokosiński 
- [ ] all checked
-   10-27 11:34:11    679adb0064 Compare Avro SerDe class name case sensitively
-   10-27 14:58:18    3d7bf899e1 Fix HiveClientConfig to use bean validation NotNull
-   10-27 14:58:18    a7cd63da7f Remove unnecessary usage of javax.validation.constraints.NotNull
-   10-27 14:58:18    f2d966d060 Remove usage of javax.annotation.Nonnull
-   10-28 13:33:07    52cff69bd9 Make sure that file based access control have coherent config properties
-   10-28 13:33:07    61cf182713 Unify json parsing in file based access controls
-   10-28 13:33:07    cd0f7ff342 Implement forwarding access controls
-   10-28 13:33:07    dc60abd91a Do not wrap RuntimeExceptions in RuntimeException
-   10-28 13:33:07    e3fcb318d9 Add example access-control.properties
-   10-28 13:33:07    e748ffe89a Make FileBased*AccessControls to be refreshable
-   10-29 01:27:33    cdbf95b99b Remove insert_const_special_char product test
-   10-31 06:48:49    031c4a9f53 Update tempto to 1.50
-   11-09 02:29:21    4b445d125b Use ForwardingConnectorAccessControl for PartitionsAwareAccessControl
-   11-09 02:51:55    d1df27f4d2 Do not fail glue tests on codestyle errors
-   11-09 06:25:40    f1ed60dfb8 Use Optional for nullable field in HiveConnectorFactory
## Haozhun Jin          
- [ ] all checked
-  11-01 17:22:40    8f79858a20 Rename NestedLoopJoinPagesBridge to NestedLoopJoinBridge
-  11-02 12:28:11    353eceed9b Fix memory leak for failed queries
## Jiexi Lin            
- [ ] all checked
-  11-06 14:18:50    5e79429186 Create quarantine file only once on recovery corruption
## Karol Sobczak        
- [ ] all checked
-  10-26 23:56:10    f45999748f Restore removed comment
-  10-26 23:57:19    b14d669b92 Execute LimitPushDown before WindowFilterPushDown
-  10-26 23:57:19    b8c1607654 Add feature toggle for TopNRowNumber optimization
-  10-26 23:57:49    186756c1bd Add support for FunctionCall to ScalarStatsCalculator
-  10-26 23:57:49    48fe4a6938 Add support for TINYINT, SMALLINT to LiteralEncoder
-  10-26 23:57:49    ab15ac9e87 Fix exception message
-  10-26 23:58:21    1b872014f7 Extract PredicatePushDown#areExpressionsEquivalent method
-  10-26 23:58:21    2b2f4b8331 Do not create new join node when equi criteria and filter are same
-  10-26 23:58:21    380eff89ed Fix null handling in expression equivalence
-  10-26 23:58:21    92b8a8889c Do not create new filter node when predicates are equivalent
-  10-26 23:58:21    ff57057dda Do not create new projection when output symbols did not change
-  10-26 23:58:52    5d4d79ac9d Remove redundant private keywords
-  10-26 23:58:52    76d3b44435 Add WorkProcessor#yielding method
-  10-26 23:58:52    c528f85ea4 Use WorkProcessor#yielding in MergeSortedPages#buildPage
-  10-29 02:41:42    7065180c52 Add @Nullable annotation and requireNonNull
-  10-29 02:41:42    bd63cab3d0 Add requireNonNull check
-  10-30 00:53:05    8bc6971e6a Make all metastore objects serializable
-  10-30 00:53:05    911e1fb3c2 Allow empty partitionName value when serializing
-  10-30 00:53:05    95c8715fa4 Add proper equals, hashCode, and toString for metastore objects
-  10-30 00:53:05    c178355592 Implement RecordingHiveMetastore
-  10-30 00:53:05    ccdc098ee3 Add @Immutable annotation to all metastore objects
-  10-30 00:53:05    f6f3bc4768 Extract nested classes from CachingHiveMetastore
-  10-30 12:40:00    12ab5432cb Assure output page is loaded in PageSourceOperator
-  10-30 15:03:58    4a82e07f2a Rename misleading method name
-  10-31 06:05:44    59ffc45125 Remove redundant isSingleNode check
## Maria Basmanova      
- [ ] all checked
-  10-27 14:17:02    53bc1e3e8b Add distribution type to semi join query plan
## Nezih Yigitbasi      
- [ ] all checked
-  10-30 22:48:24    2e85c57970 Add config for setting the ACL type for S3 uploads
-  10-31 11:08:33    ea69f54043 Add space after hyphens for ngrams example
-  10-31 16:02:23    279eb7e2af Simplify the use of Optionals
-  10-31 16:02:23    33a6240a0a Remove unnecessary type arguments
-  10-31 16:02:23    a7340ccc0c Use foreach instead of for loop
-  10-31 16:02:23    ce9459646f Cleanup null/empty checks
## Piotr Findeisen      
- [ ] all checked
-  10-27 00:20:04    8431a5d951 Remove legacy TransformCorrelatedSingleRowSubqueryToProject optimizer
-  10-27 00:20:39    8c9df8268c Update documentation for guarantees provided by startMemoryRevoke()
-  10-27 00:20:39    cbe66e90a2 Update documentation for startMemoryRevoke()
-  10-27 00:20:56    8937e13324 Fix typo in TransactionMetadata method name
-  10-27 13:39:33    bdea527092 Make exception handling in SqlQueryManager more resilient
-  10-29 15:20:59    fa992dedd7 Remove misleading test case for findEndPosition
-  10-29 15:20:59    ff0df17c5c Simplify WindowOperator.findEndPosition
-  10-30 01:07:15    81f0a55510 Fix typo in json property in ThreadResource
-  10-30 01:50:03    74ae28d191 Remove hive_connector test group
-  10-30 05:37:21    08dab3f7f2 Change JVM time zone in tests to better test corner cases
-  10-30 05:37:21    297f6ef082 Simplify SqlTimestamp construction in tests
-  10-30 05:37:21    79f030d3b1 Clarify expected behavior after representation fixed in JDBC
-  10-30 05:37:21    bf53b82aac Change default session zone in tests to non-UTC
-  10-30 05:37:21    c43282e8b3 Add more DATE/TIME related tests
-  10-30 05:37:21    f0de22a682 Upgrade Presto hive-apache to 1.2.2-2
-  10-31 04:47:35    582bb48ab0 Rename WorkProcessorUtils.yielding to match WorkProcessor.yielding
-  10-31 04:47:35    5f958066d4 Fix rawtypes use in ProcessorState
-  10-31 04:47:35    9658f5901c Remove ProcessorState constructor preconditions
-  10-31 04:47:35    b0760b6fe1 Separate TransformationState and ProcessState
-  11-01 07:08:58    56360c397c Remove unused field in PartitionedOutputOperator
-  11-06 00:05:08    fcfef105e9 Fix typo in the regexp_like documentation
## Raghav Sethi         
- [ ] all checked
-  11-02 09:50:52    5540b85b31 Fix link to Web UI home page
-  11-02 09:50:52    c954eda4ab Fix filtering in query list when sort order is changed
## Timothy Meehan       
- [ ] all checked
-  10-28 22:46:28    c53b5dd1d3 Add UDF ngrams
-  10-28 22:46:28    ff82d4c25d Fix bug in DictionaryBlock getRegion
## Wenlei Xie           
- [ ] all checked
-  10-27 20:45:27    147ce4a2bb Limit the size of page chunk for OrcWriter
-  11-07 14:28:48    87f7e51c99 Support reading DWRF flat map
## yingsu00             
- [ ] all checked
-  11-07 23:00:18    23de11f7d3 Lazily build hashtable for MapBlock
-  11-07 23:05:25    290f5be2e2 Replace floorMod with faster reduction function

**Comment by kokosing** [comment]
```
Security
--------
 * Add ``security.refresh-period`` to file based access controllers that allows to reload their configuration
```

**Comment by raghavsethi** [comment]
```rst
Web UI Changes
----------------
* Fix bug that prevented non-default sort orders from being selected in query list view.

**Comment by mbasmanova** [comment]
```
General
-------------
* Print distribution type for semi joins in explain plans
```

**Comment by tdcmeehan** [comment]
```
General
---------
* Add ngrams function to generate the n-grams from an array
* Add qdigest type and associated functions
```

**Comment by ggreg** [comment]
@electrum can you summarize your changes?

**Comment by ggreg** [comment]
@sopel39 are you done with your changes summary?

**Comment by electrum** [comment]
```
General Changes
---------------

* Improve error message for broken :func:`array_sort` comparator.

JDBC Driver Changes
-------------------

* Clear update count after calling ``Statement.getMoreResults()``.

Hive Connector Changes
----------------------

* Retry on metastore connection errors.

Verifier Changes
----------------

* Handle SQL execution timeouts while rewriting queries.

**Comment by sopel39** [comment]
```
General Changes
---------------

* Fix planning failures when lambdas are used in join filter expression.
```

@ggreg added one more note

**Comment by ggreg** [comment]
Release notes #11938
